examples/distributed/example_sp_all2all_attention_intra_node.py

```
@@ -0,0 +1,336 @@
+    try:
+        assert rank == local_rank and num_ranks == num_local_ranks, "only support single node for now"
+
+        seqlens_q = args.seqlens_q
+        seqlens_k = args.seqlens_k
+        assert len(seqlens_q) == batch_size and len(seqlens_k) == batch_size
+        assert q_head % num_ranks == 0, "q_head should be divisible by world size"
+        assert kv_head % num_ranks == 0, "kv_head should be divisible by world size"
+        for s in seqlens_q + seqlens_k:
+            assert s % num_ranks == 0, "all2all requires per-batch sequence length divisible by world size"
+
+        cu_seqlens_q_list = [0] + list(accumulate(seqlens_q))
```

Ruff (0.15.9)

[warning] Consider `[0, *list(accumulate(seqlens_q))]` instead of concatenation

Replace with `[0, *list(accumulate(seqlens_q))]`

(RUF005)